src/Neuron/Zettelkasten/Route.hs

```diff
@@ -27,14 +28,27 @@ import qualified Text.URI as URI
 data Route store graph a where
   Route_Redirect :: ZettelID -> Route ZettelStore ZettelGraph ZettelID
   Route_ZIndex :: Route ZettelStore ZettelGraph ()
+  Route_Search :: {searchTerms :: Maybe String, searchTags :: [Text]} -> Route ZettelStore ZettelGraph ()
```

Comment:
This looks semantically off. The constructor arguments should contain only the information needed to determine the route .html filename. As this is `search.html` in this case, there should be no constructor arguments at all. Now, the "data" required to *render* the route is to be specified in the GADT type parameters (`store`, `graph` and `a`). Looking at `renderSearch`, it doesn't look like you need anything extra to render the search page. So this should really be:

```haskell
Route_Search :: Route ZettelGraph ZettelGraph ()
```